### PR TITLE
[WIP] Add symfony 3.0 support

### PR DESCRIPTION
**Pull request by rande**

### CODE REVIEW

File: Form/Type/BooleanType.php [hunk 1]
```diff
@@ -89,6 +89,14 @@ public function getParent()
...
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getBlockPrefix()
+    {
         return 'sonata_type_boolean';
     }
-}
+}
```
Comment (Soullivaneuh) [inline Form/Type/BooleanType.php]: newline

File: Resources/config/core.xml [hunk 2]
```diff
@@ -7,5 +7,15 @@
...
+            <argument type="collection" />
+            <argument type="collection" />
+            <argument type="collection" />
+            <argument type="collection" />
+            <argument type="collection" />
+            <argument type="collection" />
+        </service>
     </services>
-</container>
+</container>
```
Comment (Soullivaneuh) [inline Resources/config/core.xml]: newline

**Comment by rande** [comment]
Waiting for https://github.com/symfony/symfony/pull/16615 to be merged

**Comment by rande** [comment]
@Ener-Getick I just integrate some of your changes.

**Comment by Ener-Getick** [comment]
:+1:

File: Resources/config/core.xml (hunk 2 above)
Comment (rande) [inline Resources/config/core.xml]: There is no newline, do you want one ?

File: Tests/Form/Type/DoctrineORMSerializationTypeTest.php [hunk 3]
```diff
@@ -63,7 +65,12 @@ public function testBuildForm()
         $metadataFactory = $this->getMetadataFactoryMock();
         $registry = $this->getRegistryMock();
 
-        $type = new DoctrineORMSerializationType($metadataFactory, $registry, 'form_type_test', $this->class, 'serialization_api_write');
+        if (Kernel::MAJOR_VERSION < 3) {
```
Comment (Ener-Getick) [inline Tests/Form/Type/DoctrineORMSerializationTypeTest.php]: The problem with this check is that the deprecations are not removed for sf 2.8 so the Travis build will fail... :-/

File: Resources/config/core.xml (hunk 2 above)
Comment (Soullivaneuh) [inline Resources/config/core.xml]: Yes sir. :-)

**Comment by OskarStark** [comment]
:+1:

**Pull request merged**